src/Symfony/Component/Mailer/Bridge/Mailgun/Transport/MailgunApiTransport.php

```diff
@@ -130,17 +130,7 @@ private function getPayload(Email $email, Envelope $envelope): array
                 continue;
             }
 
-            // Check if it is a valid prefix or header name according to Mailgun API
-            $prefix = substr($name, 0, 2);
-            if (\in_array($prefix, ['h:', 't:', 'o:', 'v:']) || \in_array($name, ['recipient-variables', 'template', 'amp-html'])) {
-                $headerName = $name;
-            } else {
-                // fallback to prefix with "h:" to not break BC
-                $headerName = 'h:'.$name;
-                trigger_deprecation('symfony/mailer', '5.1', 'Not prefixing the Mailgun header name with "h:" is deprecated. Use header name "%s" instead.', $headerName);
```

Comment:
I wonder if we should keep the check and turn the deprecation into an exception instead.

Reply:
Yep, I also hesitate here

Reply:
I strongly think that we should actually *revert* that deprecation as a bugfix (see my review on the PR introducing it, as this deprecation makes it impossible to write code that sends an email with headers, while working with both the mailgun transport and other transport without changes)